### PR TITLE
Map Maker Changes

### PR DESCRIPTION
**Pull request by moreavy**

### CODE REVIEW

**Review by JostP — APPROVED** [review]

File: mods/ctf/ctf_map/map_maker/README.md [hunk 1]
```diff
@@ -110,12 +109,21 @@ If you want to tell more infomation, you can use:
...
 
-```lua
+```properties
 treasures = default:pick_steel,0.5,5,1,10;shooter:shotgun,0.04,2,1;shooter:grenade,0.08,2,1
 ```
 
 (See [here](../../../other/treasurer/README.md) to understand the magic numbers)
 
+#### `initial_stuff`
+The `initial_stuff` are the items given to everyone at their (re)spawn. A pickaxe and a torch is promoted to be given in the maps `initial_stuff`.
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:119]:
```suggestion
`initial_stuff` are the items given to players at their (re)spawn. At least pickaxe and some torches should be given in the map's `initial_stuff`.
```

File: mods/ctf/ctf_map/map_maker/README.md [hunk 2]
```diff
@@ -110,12 +109,21 @@ If you want to tell more infomation, you can use:
...
+
+An example `initial_stuff` value that registers a stone pickaxe, 30 cobblestones and 5 torches is given below.
+
+```properties
+initial_stuff = default:pick_stone,default:cobble 30,default:torch 5
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:124]:
```suggestion
An example of `initial_stuff` value that registers a stone pickaxe, 30 cobblestones, 5 torches and a pistol is given below.

```properties
initial_stuff = default:pick_stone,default:cobble 30,default:torch 5,shooter:pistol
```

**Review by JostP — APPROVED** [review]

**Comment by xenonca** [comment]
Could add a note in the README about the `z=0` definition not working until it's fixed [#706 (comment)](https://github.com/MT-CTF/capturetheflag/issues/706#issuecomment-761765562), [#157 (maps)](https://github.com/MT-CTF/maps/issues/157)

**Comment by moreavy** [comment]
> Could add a note in the README about the `z=0` definition not working until it's fixed [#706 (comment)](https://github.com/MT-CTF/capturetheflag/issues/706#issuecomment-761765562), [#157 (maps)](https://github.com/MT-CTF/maps/issues/157)

Shall I also change the Map Maker Mod by removing the option `X=0`, `Z=0` and just use the `Place Barrier` Button? I can make those changes in the formspec.

**Comment by moreavy** [comment]
Removed `X=0` option.

File: mods/ctf/ctf_map/map_maker/README.md [hunk 3]
```diff
@@ -133,7 +140,7 @@ Six images which should be in map's folder.
 * `skybox_5.png` - south
 * `skybox_6.png` - north
 
-You have to include skybox license in `license` in `.conf` file. We can only accept Free Software licenses, e.g. `CC0`, `CC BY 3.0`, `CC BY 4.0`, `CC BY-SA 3.0`, `CC BY-SA 4.0`.
+You have to include skybox license in `license` in `map.conf` file. We can only accept Free Software licenses, e.g. `CC0`, `CC BY 3.0`, `CC BY 4.0`, `CC BY-SA 3.0`, `CC BY-SA 4.0`. When you include the skybox images in the map's directory, run the `update.sh` file in the `games/capturetheflag/` folder.
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:143]:
```suggestion
You have to include skybox license in `license` in `map.conf` file. We can only accept Free Software licenses, e.g. `CC0`, `CC BY 3.0`, `CC BY 4.0`, `CC BY-SA 3.0`, `CC BY-SA 4.0`.
Before you test your skybox images in local CTF game, run the `update.sh` file in the `games/capturetheflag/` folder.
```

File: mods/ctf/ctf_map/map_maker/README.md [hunk 4]
```diff
@@ -35,9 +35,8 @@ There are multiple ways do this, this is the simplest in most cases.
 
 ### 4. Place barriers
 
-* Set the middle barrier direction. The barrier is a plane defined by a co-ordinate = 0.
-  If the barrier is X=0, then it will placed with every node of the barrier having X=0.
-  If the barrier is Z=0, then it will placed with every node of the barrier having Z=0.
+* The barrier is a plane defined by Z=0.
+* The barrier will be placed having the Z co-ordinate as 0.
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:39]: I don't think that removing these lines is a good idea. We will need them later after map maker mod is fixed.

I would suggest  ~~crossing~~ these lines if possible, and then adding line that tells why X=0 doesn't work.

Comment (JostP) [reply mods/ctf/ctf_map/map_maker/README.md:39]: And modifying map maker's `gui.lua` isn't good idea too. I think adding just line that tells why X=0 doesn't work to documentation would be fine.

Comment (moreavy) [reply mods/ctf/ctf_map/map_maker/README.md:39]: why @JostP ?

Comment (JostP) [reply mods/ctf/ctf_map/map_maker/README.md:39]: > We will need them later after map maker mod is fixed.


> Whoever will try to fix map maker mod, he will need not-modified code.

Comment (moreavy) [reply mods/ctf/ctf_map/map_maker/README.md:39]: You mean you need `X=0`?

File: mods/ctf/ctf_map/map_maker/README.md [hunk 5]
```diff
@@ -133,18 +140,19 @@ Six images which should be in map's folder.
 * `skybox_5.png` - south
 * `skybox_6.png` - north
 
-You have to include skybox license in `license` in `.conf` file. We can only accept Free Software licenses, e.g. `CC0`, `CC BY 3.0`, `CC BY 4.0`, `CC BY-SA 3.0`, `CC BY-SA 4.0`.
+You have to include skybox license in `license` in `map.conf` file. We can only accept Free Software licenses, e.g. `CC0`, `CC BY 3.0`, `CC BY 4.0`, `CC BY-SA 3.0`, `CC BY-SA 4.0`.
+Before you test your skybox images in local CTF game, run the `update.sh` file in the `games/capturetheflag/` folder.
 
 You can find some good skyboxes with suitable licenses at [https://opengameart.org](https://opengameart.org/art-search-advanced?field_art_tags_tid=skybox).
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:147]:
```suggestion
You can find some good skyboxes with suitable licenses at [opengameart.org](https://opengameart.org/art-search-advanced?field_art_tags_tid=skybox) or [www.humus.name](https://www.humus.name/index.php?page=Textures).

```

**Comment by moreavy** [comment]
@JostP Informed that the `X=0` option not working.

File: mods/ctf/ctf_map/map_maker/README.md [hunk 6]
```diff
@@ -35,8 +35,9 @@ There are multiple ways do this, this is the simplest in most cases.
 
 ### 4. Place barriers
 
-* The barrier is a plane defined by Z=0.
-* The barrier will be placed having the Z co-ordinate as 0.
+* The barrier is a plane defined by co-ordinate (=0).
+* If you chooze `X=0` The barrier will be placed having the X co-ordinate as 0. But from a few months, the `X=0` co-ordinate creates bugs and errors. It's better if you choose `Z=0` for creating your map.
+* If you chooze `Z=0` The barrier will be placed having the Z co-ordinate as 0.
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:40]:
```suggestion
* If you choose `X=0` The barrier will be placed having the X co-ordinate as 0. But from a few months, the `X=0` co-ordinate creates bugs and errors. It's better if you choose `Z=0` for creating your map.
* If you choose `Z=0` The barrier will be placed having the Z co-ordinate as 0.
```

**Comment by moreavy** [comment]
Anything more to do?

**Comment by JostP** [comment]
Does this PR fix map maker problems?

**Comment by moreavy** [comment]
No, It just tells (in the README.md) that the rotation `X=0` is not working.

File: mods/ctf/ctf_map/map_maker/gui.lua [hunk 7]
```diff
@@ -55,7 +57,8 @@ end
 function map_maker.show_progress_formspec(name, text)
 	minetest.show_formspec(name, "ctf_map:progress",
 		"size[6,1]bgcolor[#080808BB;true]" ..
-		"label[0,0;" ..
+		default.gui_bg ..
+		default.gui_bg_img .. "label[0,0;" ..
 		minetest.formspec_escape(text) .. "]")
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/gui.lua:62]: What are this file changes?

File: mods/ctf/ctf_map/map_maker/README.md [hunk 8]
```diff
@@ -35,9 +35,9 @@ There are multiple ways do this, this is the simplest in most cases.
 
 ### 4. Place barriers
 
-* Set the middle barrier direction. The barrier is a plane defined by a co-ordinate = 0.
-  If the barrier is X=0, then it will placed with every node of the barrier having X=0.
-  If the barrier is Z=0, then it will placed with every node of the barrier having Z=0.
+* The barrier is a plane defined by co-ordinate (=0).
+* If you choose `X=0` The barrier will be placed having the X co-ordinate as 0. But from a few months, the `X=0` co-ordinate creates bugs and errors. It's better if you choose `Z=0` for creating your map.
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:39]:
```suggestion
* If you choose `X=0` the barrier will be placed having the X co-ordinate as 0. But from a few months, the `X=0` co-ordinate creates bugs and errors. It's better if you choose `Z=0` for creating your map.
```

File: mods/ctf/ctf_map/map_maker/gui.lua (hunk 7 above)
Comment (moreavy) [reply mods/ctf/ctf_map/map_maker/gui.lua:62]: These are change in The GUI, not in the rotation. These changes may not matter in this _formspec_ versions but matter if the _formspec_ version is upgraded.

File: mods/ctf/ctf_map/map_maker/README.md [hunk 9]
```diff
@@ -133,18 +141,18 @@ Six images which should be in map's folder.
...
 ## Editing exported map
 
 The easiest way to edit exported maps is the following:
 * Create a world using `singlenode` mapgen. Enable `WorldEdit` and `ctf_map` mod,
-* Go in the world's folder, create a folder named `schems`, and place the `.mts` file inside,
+* Go in the world's folder, create a folder named `schems`, and place the `map.mts` file inside,
 * Start the game, `/grantme all` and enable `fly` (there is no ground in singlenode mapgen),
 * Do `//1` to set the position where you will generate the map,
-* Do `//mtschemplace yourschematic` (where `yourschematic` is the name of the mts file without `.mts`).
+* Do `//mtschemplace map`.
```
Comment (JostP) [inline mods/ctf/ctf_map/map_maker/README.md:155]:
```suggestion
* Do `//mtschemplace map` (where `map` is the name of the mts file without `.mts`).
```

File: mods/ctf/ctf_map/map_maker/gui.lua (hunk 7 above)
Comment (JostP) [reply mods/ctf/ctf_map/map_maker/gui.lua:62]: If it works then you probably don't have to remove.

**Review by JostP — APPROVED** [review]

**Comment by Panquesito7** [comment]
The repository fork has been deleted. @moreavy I suggest you make a new PR.
@LoneWolfHT Once new PR has been made, I think this can be merged.

**Comment by LoneWolfHT** [comment]
I apparently don't need the fork to merge it

**Pull request merged**